### PR TITLE
Include the section 'Configure Eclipse to Edit Checker Framework' und…

### PR DESCRIPTION
**Pull request by h82lin**
Contains the instructions for beginners to configure Eclipse in order to edit Checker Framework @wmdietl

### CODE REVIEW

**Comment by h82lin** [comment]
@wmdietl  Hi Professor, I tried to PR what I added to the troubleshooting.tex file at first and Travis CI told me there were formatting errors. I later tried the command "ant reformat" and it changed the formatting of a few files. The change was pushed again and its now giving me errors regarding failure to test "sortwarnings/Main.java". Do you know any information about this error? Thanks!

File: docs/manual/troubleshooting.tex [hunk 1]
```diff
@@ -968,6 +968,44 @@ \subsection{Build the Checker Framework Manual (this document)\label{building-ma
 \end{enumerate}
 
 
+\subsection{Configure Eclipse to Edit Checker Framework\label{building-eclipse}}
```
Comment (wmdietl) [inline docs/manual/troubleshooting.tex]: `Edit` -> `edit the`

File: docs/manual/troubleshooting.tex [hunk 2]
```diff
@@ -968,6 +968,44 @@ \subsection{Build the Checker Framework Manual (this document)\label{building-ma
...
+\begin{enumerate}
+\item
+Install Eclipse from the Eclipse official website.
```
Comment (wmdietl) [inline docs/manual/troubleshooting.tex]: Change to: `... from the official Eclipse...`
Add a URL.

File: docs/manual/troubleshooting.tex [hunk 3]
```diff
@@ -968,6 +968,44 @@ \subsection{Build the Checker Framework Manual (this document)\label{building-ma
...
+
+\item
+Once installation is complete, run Eclispe and set the working directory to jsr308.
```
Comment (wmdietl) [inline docs/manual/troubleshooting.tex]: Should the `jsr308` be `$JSR308`?
Use `\<>` for in-line terminal font, throughout your changes.

File: docs/manual/troubleshooting.tex [hunk 4]
```diff
@@ -968,6 +968,44 @@ \subsection{Build the Checker Framework Manual (this document)\label{building-ma
...
+
+\item Try locating the "plume-lib" file in jsr308. If plume-lib already exists, ignore this step. Otherwise, depending on the organization you are working from(typetools, opprop, or pascaliUWat), git-clone plume-lib by: (change "typetools" in git-clone URL below to desired organization)
```
Comment (wmdietl) [inline docs/manual/troubleshooting.tex]: I would make this a separate preparation step before the steps within Eclipse.
You also need to do a `make all` once to generate a few files.

Comment (wmdietl) [inline docs/manual/troubleshooting.tex]: Break long lines. That makes commenting on specific parts easier and future changes less likely to conflict.

Comment (wmdietl) [inline docs/manual/troubleshooting.tex]: Multiple organizations aren't mentioned anywhere else in this section, so I wouldn't mention it here.

File: docs/manual/troubleshooting.tex [hunk 5]
```diff
@@ -968,6 +968,44 @@ \subsection{Build the Checker Framework Manual (this document)\label{building-ma
...
+Enter the main eclipse working screen and under "File" tab, select Import -> General -> Existing Projects into Workspace. Now that the Import Projects window appear, tick "Select Root Directory" and browse for these files you have previously included in jsr308:
+\begin{itemize}
+\item annotation-files-utilities, asmx, checker, dataflow, framework, javacutil, jsr308-langtools, scene-lib, stubparser, plume-lib. (Be sure all these files were git-cloned from the same organization, either typetools, opprop, or pascaliUWat)
+\item Import the above individually. There might be more files that need to be imported depending on which Java file you would like to edit.
+\end{itemize}
+
+\item
+After all projects have been imported, error messages should not appear. Please resolve them if there are any.
+
+\item A common error encountered at this point is that tools.jar file cannot be found. A way to resolve this issue is by creating a softlink of \<JAVA\_HOME> in the plume-lib directory. This is because tools.jar is just a lib jar in the \<JAVA\_HOME> directory.
```
Comment (wmdietl) [inline docs/manual/troubleshooting.tex]: Put this together with cloning plume-lib as a preparation step on the command line.

File: docs/manual/troubleshooting.tex [hunk 6]
```diff
@@ -968,6 +968,44 @@ \subsection{Build the Checker Framework Manual (this document)\label{building-ma
...
+\begin{itemize}
+\item
+Command to create softlink:
+\begin{Verbatim}
+cd $JSR308/plume-lib
+ln -s /usr/lib/jvm/java-8-openjdk-amd64 jdk
+\end{Verbatim}
+\item After a softlink is created, add tools.jar to the plume-lib build path. Do so by right clicking plume-lib under "Package Explorer" -> Build Path -> Configure Build Path. After the "Properties" window appear, under "Libraries" click "Add JARs" and add tools.jar from the softlink.
```
Comment (wmdietl) [inline docs/manual/troubleshooting.tex]: Why do you need to do this? `tools.jar` is already in `plume-lib/.classpath`. If you do these steps before going into Eclipse, this shouldn't be needed.

File: docs/manual/troubleshooting.tex (hunk 4 above)
Comment (h82lin) [inline docs/manual/troubleshooting.tex]: So should I include a part before this that briefly introduces the different organisations?

File: docs/manual/troubleshooting.tex (hunk 6 above)
Comment (h82lin) [inline docs/manual/troubleshooting.tex]: Yes, thanks for pointing this out. This last part shouldn't be needed

File: docs/manual/troubleshooting.tex (hunk 4 above)
Comment (wmdietl) [inline docs/manual/troubleshooting.tex]: That might be a good idea. However, most people will use typetools. Let's not add further complication to the manual.

**Comment by wmdietl** [comment]
Can you pull in upstream changes, both into master and into your branch, to remove the reformatting issues?

**Comment by topnessman** [comment]
Duplicate with #41

**Pull request closed without merging**